### PR TITLE
Claim RAD UI

### PR DESCRIPTION
**Pull request by philipstanislaus**

### CODE REVIEW

**Review by offerijns — APPROVED** [review]
This looks really great Philip!

**Comment by philipstanislaus** [comment]
Thanks for your feedback @offerijns , merging this now and will address all your points in another PR.

File: tinlake-ui/components/LoadingValue/index.tsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
 import * as React from 'react'
 import styled from 'styled-components'
 
-interface LoadingValueProps {
+interface PropsWithChildren {
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/components/LoadingValue/index.tsx:4]: Good point, changed in a9672f2cd7040817eaa9e0ffb7a922c4c9ca00ac

File: tinlake-ui/components/LoadingValue/index.tsx [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+  maxWidth?: number
   alignRight?: boolean
 }
 
+interface PropsWithRender {
+  done: boolean
+  height?: number
+  maxWidth?: number
+  alignRight?: boolean
+  render: () => React.ReactNode
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/components/LoadingValue/index.tsx:17]: Using a function allows conditional evaluation of the arguments to that function. I added it here so we can actually use values we check in the `done` property, without needing to check for undefined/null/loading in the arguments again.

Created a small codesandbox for you to play around with – feel free to ping me on Slack to show you what I mean: https://codesandbox.io/s/dreamy-wright-4g97l?file=/src/App.js

File: tinlake-ui/components/WalletRewards/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import styled from 'styled-components'
+import { load, UserRewardsState } from '../../ducks/userRewards'
+import { toDynamicPrecision } from '../../utils/toDynamicPrecision'
+
+export const WalletRewards = ({ address }: { address: string }) => {
+  const dispatch = useDispatch()
+  const userRewards = useSelector<any, UserRewardsState>((state: any) => state.userRewards)
+  React.useEffect(() => {
+    if (address) {
+      dispatch(load(address))
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/components/WalletRewards/index.tsx:16]: Changed

File: tinlake-ui/containers/ClaimRewards/index.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  return (
+    <>
+      <Box pad={{ horizontal: 'medium', bottom: 'medium' }}>
+        {unclaimed && !unclaimed?.isZero() ? (
+          <>
+            {(status === null || status === 'unconfirmed' || status === 'failed' || status === 'pending') && (
+              <>
+                🎉 You can claim {toDynamicPrecision(baseToDisplay(unclaimed, 18))} RAD rewards. Claim now to stake RAD
+                and participate in on-chain governance.
+                {unclaimed!.gt(data?.totalEarnedRewards || new BN(0)) && (
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/containers/ClaimRewards/index.tsx]: Good catch, changed in: 29b358996510c8ba7db15535753837ebf9aae419

File: tinlake-ui/ducks/centChainWallet.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+const INJECT_ACCOUNTS = 'tinlake-ui/cent-chain-wallet/INJECT_ACCOUNTS'
+
+// Errors
+const ERR_NO_EXT_OR_NOT_AUTH = 'No extension or not authorized'
+
+export interface InjectedAccount {
+  name: string | undefined
+  source: string
+  addrInjected: string // this is the address as received from the extension. It seems that it always comes as the default Substrat SS58 encoded address, no matter what is set on the extension.
+  addrCentChain: string // this is the same public key as in addrInjected, but transformed to the Cetnrifuge Chain network prefix.
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/ducks/centChainWallet.ts:21]: Good find

File: tinlake-ui/containers/ClaimRewards/index.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+          onClick={claim}
+        />
+      </RewardStripe>
+    </>
+  )
+}
+
+export default ClaimRewards
+
+const RewardStripe = ({ unclaimed, children }: React.PropsWithChildren<{ unclaimed: BN | null }>) => (
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/containers/ClaimRewards/index.tsx:171]: Done in ff614550f46bcc77760716aed7f2bb55035c33a7

File: tinlake-ui/services/apollo/index.ts [hunk 7]
```diff
@@ -283,6 +286,113 @@ class Apollo {
...
+        earned: new BN(link.rewardsAccumulated),
+        claimable: null,
+        claimed: null,
+      })),
+    }
+
+    return transformed
+  }
+
+  // async getRewardsByUserToken(user: string) {
```
[the remark this answers is not in the document]
Comment (philipstanislaus) [reply tinlake-ui/services/apollo/index.ts:382]: Yes, done in e601da6e2e4367ccf2b8ea177ca0e21a13561198

**Pull request merged**